app/views/components/_autocomplete.html.erb

```diff
@@ -1,27 +1,18 @@
 <%
   id ||= "autocomplete-#{SecureRandom.hex(4)}"
-  type ||= nil
   error_id = "error-#{SecureRandom.hex(4)}"
   error_items ||= []
-  hint ||= nil
-  width ||= nil
   aria = error_id if error_items.any?
   data_attributes ||= {}
   track_category ||= nil
   track_label ||= nil
   select ||= {}
-  input ||= {}
-  search ||= false
-  jsonly ||= false
 
   root_classes = %w(app-c-autocomplete govuk-form-group)
-  root_classes << "app-c-autocomplete--#{width}" if width
-  root_classes << "app-c-autocomplete--search" if search
-  root_classes << "app-js-only" if jsonly
   root_classes << "govuk-form-group--error" if error_items.any?
 %>
 
-<%= tag.div class: root_classes, data: data_attributes.merge(module: "autocomplete", "autocomplete-type": type) do %>
+<%= tag.div class: root_classes, data: data_attributes.merge(module: "autocomplete") do %>
```

Comment:
One thing that i noticed about this is it overrides any existing modules? Is this the correct behaviour?